service/src/main/java/crawlercommons/urlfrontier/service/rocksdb/RocksDBService.java

```diff
@@ -960,15 +962,17 @@ public URLItem next() {
                     final int pos2 = schedulingKey.indexOf('_', pos1 + 1);
                     final int pos3 = schedulingKey.indexOf('_', pos2 + 1);
 
-                    fromEpoch = Long.parseLong(schedulingKey.substring(pos2 + 1, pos3));
+                    fromEpoch = Long.parseLong(schedulingKey, pos2 + 1, pos3, 10);
 
                     try {
                         info = URLInfo.parseFrom(scheduled);
                     } catch (InvalidProtocolBufferException | NumberFormatException e) {
                         LOG.error(e.getMessage(), e);
                     }
                 } else {
-                    LOG.debug("no schedule for {}", currentKey);
+                    if (LOG.isDebugEnabled()) {
```

Comment:
I don't think there is any benefit in checking the level of the log. This is worth doing only if an expensive operation is in the scope, here we just log and the string concatenation should not happen if the level is not debug

Reply:
Indeed, not needed anymore with SL4J (was useful with Log4J).